src/tests/ftest/util/fault_config_utils.py

```diff
@@ -211,34 +218,79 @@ class FaultInjectionFailed(Exception):
-
-        fault_config = []
-        if fault_list is not None:
-            for fault in fault_list:
-                fault_config.append(FAULTS[fault])
-        if on_the_fly_fault is not None:
-            fault_config.append(on_the_fly_fault)
-        yaml.dump({'fault_config': fault_config}, outfile,
-                  default_flow_style=False, allow_unicode=True)
-
-    return fi_config
+    def __init__(self):
+        super().__init__()
+        self._hosts = []
+        self._fault_file = None
+        self._test_dir = None
+        self._fault_list = []
+
+    def write_fault_file(self, on_the_fly_fault=None):
+        """ Write out a fault injection config file.
+
+        Args:
+            on_the_fly_fault --a fault dictionary that isn't predefined
+        """
+        if self._fault_list is None and on_the_fly_fault is None:
+            raise FaultInjectionFailed("bad parameters")
+
+        fi_config = os.path.join(self._test_dir, "fi.yaml")
+
+        with io.open(fi_config, 'w', encoding='utf8') as outfile:
+            yaml.dump({'seed': '123'}, outfile, default_flow_style=False, allow_unicode=True)
+            fault_config = []
+            if self._fault_list is not None:
+                for fault in self._fault_list:
+                    fault_config.append(FAULTS[fault])
+            if on_the_fly_fault is not None:
+                fault_config.append(on_the_fly_fault)
+            yaml.dump({'fault_config': fault_config}, outfile,
+                        default_flow_style=False, allow_unicode=True)
+
+        os.environ["D_FI_CONFIG"] = fi_config
+
+        self._fault_file = fi_config
+
+    def start(self, fault_list, test_dir):
+        """Create the fault injection file to inject DAOS faults.
+
+        Args:
+            fault_list (list): List of faults to inject.
+            test_dir(str) : Path to create the fault injection file.
+        """
+        self._fault_list = fault_list
+        self._test_dir = test_dir
+        if self._fault_list:
+            # not using workdir because the huge path was messing up
+            # orterun or something, could re-evaluate this later
+            self.write_fault_file(None)
+
+    def copy_fault_files(self, hosts):
+        """Copy the fault injection file to all test hosts.
+
+        Args:
+            hosts (list): list of hosts to copy the fault injection file
+        """
+        if self._fault_list:
+            self._hosts = hosts
+            distribute_files(self._hosts, self._fault_file, self._fault_file)
+
+    def stop(self):
+        """Remove the fault injection file created during testing.
+        Returns:
+           error_list (list) : Errors during removing fault files (if any).
+        """
+        # Remove the fault injection files on the hosts.
+        error_list = []
+        rm_command = "{} rm -f {}".format(
+                        get_clush_command(self._hosts, "-S -v", True), self._fault_file)
+        try:
```

Comment:
For the situation where fault injection is being used with the `TestWithoutServer` class - where the `FaultInjection.copy_fault_files()` method will not be called - this fail because `clush` will be called witha n empty `-w` list.
```suggestion
        commands = []
        if self._host_list:
            commands.append(get_clush_command(self._hosts, "-S -v", True))
        commands.append("rm -f {}".format(self._fault_file))
        try:
            run_command(" ".join(commands), verbose=True, raise_exception=False)
```

Reply:
Done.